library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -47,10 +47,19 @@
  * @since 3.2.0
  */
 @InterfaceStability.Evolving
-public abstract class Parser<INPUT> extends BaseOperator implements Converter<INPUT, Object>,
-    ActivationListener<Context>
+public abstract class Parser<INPUT> extends BaseOperator implements Converter<INPUT, Object>
 {
   protected transient Class<?> clazz;
+  @AutoMetric
+  protected long errorTupleCount;
+  @AutoMetric
+  protected long emittedObjectCount;
+  @AutoMetric
+  protected long rollingPercentValidTuples;
+  @AutoMetric
+  protected long rollingPercentErrorTuples;
+  @AutoMetric
+  protected long incomingTuplesCount;
 
   @OutputPortFieldAnnotation(schemaRequired = true)
```

Comment:
Well, what i meant is if user doesn't want pojos he/she wont connect to this  port and would not set the class property as well. Basically user wont provide class if pojo's aren't required. But because of schemaRequired = true validation would fail